addons/binding/org.openhab.binding.edimax/src/main/java/org/openhab/binding/edimax/internal/commands/AbstractCommand.java

```diff
@@ -0,0 +1,360 @@
+
+            // Send request
+            DataOutputStream wr = new DataOutputStream(connection.getOutputStream());
+            wr.write(urlParameters.getBytes());
+            wr.close();
+
+            // Get Response
+            InputStream is = connection.getInputStream();
+            BufferedReader rd = new BufferedReader(new InputStreamReader(is));
+            StringBuilder response = new StringBuilder();
+            String line;
+            while ((line = rd.readLine()) != null) {
+                response.append(line);
+                response.append('\r');
+            }
+            rd.close();
+            return response.toString();
+        } finally {
+            if (connection != null) {
+                connection.disconnect();
+            }
+        }
+    }
+
+    /**
+     * Create initial XML tag.
+     *
+     * @param aName
+     * @return
+     */
+    protected String createStartTag(String aName) {
+        StringBuffer res = new StringBuffer();
+        res.append("<");
+        res.append(aName);
+        if ("SMARTPLUG".equals(aName)) {
+            res.append(" id=\"edimax\"");
+        } else if ("CMD".equals(aName)) {
+            if (isSet()) {
+                res.append(" id=\"setup\"");
+            } else {
+                res.append(" id=\"get\"");
+            }
+        }
+        res.append(">");
+
+        return res.toString();
+    }
+
+    /**
+     * Create end tag.
+     *
+     * @param aName
+     * @return
+     */
+    protected String createEndTag(String aName) {
+        return "</" + aName + ">";
+    }
+
+    /**
+     * Create leaf node.
+     *
+     * @param aName
+     * @return
+     */
+    protected String createLeafTag(String aName) {
+        if (isSet()) {
+            StringBuffer sb = new StringBuffer();
+            sb.append(createStartTag(aName));
+            sb.append(marshal(setValue));
+            sb.append(createEndTag(aName));
+            return sb.toString();
+        } else {
+            return "<" + aName + "/>";
+        }
+    }
+
+    /**
+     * HELPER.
+     */
+    public static final String XML_HEADER = "<?xml version=\"1.0\" encoding=\"UTF8\"?>\r\n";
+
+    /**
+     * XML Helper.
+     */
+    protected static final DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance();
+
+    /**
+     * Extract the innermost information of the response as string.
+     *
+     * @param document
+     * @param xpathExpression
+     * @return
+     * @throws ParserConfigurationException
+     * @throws SAXException
+     * @throws IOException
+     * @throws XPathExpressionException
+     */
+    protected String extractValueFromXML(String document, String xpathExpression)
+            throws ParserConfigurationException, SAXException, IOException, XPathExpressionException {
+        DocumentBuilder builder = domFactory.newDocumentBuilder();
+        ByteArrayInputStream tempIS = new ByteArrayInputStream(document.getBytes());
```

Comment:
_I would prefer proper names even in these cases, `tempIS` and `dDoc` are not really proper imho_